### ISSUE TITLE
Fixer should only fix the given file, not the entire project

### ISSUE THREAD

**Issue by cdaguerre** [FriendsOfPHP/PHP-CS-Fixer#1368]
Right now, if the `FixCommand` finds a `.php_cs`, it will fix all files returned by the `Finder`.
IMO, it should:
- Fix only files in directory and subdirectories when called on a directory
- Fix only the file it was called on when called on a file

Let me know if I should make a PR.

**Comment by SpacePossum** [comment]
:+1: passed arguments on the commandline should always override the ones in the config file IMO.
Lets see what others think though :)

**Comment by keradus** [comment]
`php-cs-fixer fix file.php` will fix that one file.
`php-cs-fixer fix dir` will fix that one dir, if dir have special configuration in `dir\.php_cs` file it will be used, if not - then not.

I can't see any problem here.

**Comment by GrahamCampbell** [comment]
:1:

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by cdaguerre** [comment]
@keradus Actually this issue is related to PR #1367.

You should be able to have a project-wide configuration in:
```
/var/www/project/.php_cs
```
And to call :
```
$ php-cs-fixer fix /var/www/project/some/bundle/file.php
```
or :
```
$ php-cs-fixer fix /var/www/project/some/bundle/
```
using the project wide config.
You probably agree you shouldn't have to put a `.php_cs` in each and every project subdirectory. ;)
I have no idea how to figure out why it won't work on HHVM though... Any clue?

#### Symfony's finder
You can't clear previous calls to `in()` in Symfony's Finder. Every time you call `->in()`, it actually merges given directories to an array of previously added dirs, as you can see [here](https://github.com/symfony/Finder/blob/master/Finder.php#L667).

#### PHP CS config
If there is a call to `$finder->in(__DIR__)` in the `Finder` returned by the `Config`, ALL project files are fixed, even if you called the `fix` command on a subdirectory.
It works as expected if the `Config`'s `Finder` does NOT scope any directories with a call to `$finder->in(__DIR__)`.

#### The problem
The problem is:
- In all `php-cs-fixer` examples, a typical `.php_cs` file includes a call to `$finder->in(__DIR__)`
- To my understanding, narrowing down in `.php_cs` as it's done in the examples isn't event necessary, as it's done [here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/Symfony/CS/Console/ConfigurationResolver.php#L354) anyway.
- Just like @SpacePossum said, IMO CLI arguments should always override higher level config.
- A simple call to `$this->dirs = array()` before [this line](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/Symfony/CS/Finder/DefaultFinder.php#L48) would have done the job, but all properties of Symfony's `Finder` are private...

#### Bottom line
I can make a PR to remove all examples of `.php_cs` including calls to `$finder->in(__DIR__)` if you want.

**Comment by SpacePossum** [comment]
Maybe I misread and I surely didn't spend much time on explaining what I think would ideal.
Lets say I have project:
`project\.php_cs -> fix dir a and b with fixers 1,2 and 3`
`project\a\.php_cs -> fix file .\foo with fixer 4`
`project\b`

when I run the fixer in:
*  `project\` 
 * with no `dir`, fix files in `/` and `b` with fixers `1,2 and 3`, `a` with fixer `4`
 * with dir `a` expected: the files in `a` to be fixed using `4` only,
 * with dir `b` expected: fixers `1,2, and 3` applied one the files in `b`.
* `project\a` expected `4` applied one files in `a`
* `project\b` expected: the defaults of the PHP-CS-Fixer applied on files in `b`

what I don't expect; 
when `project\a\.php_cs`  defines a finder `-> fix files in c`
and I run the fixer from the `project\` with dir `a` that it follows the finder defined in `a`.

Bottom line
I think the fixer should follow the configuration it finds where it is run from, if it finds others when running it should only use the `fixers` part and not the `finder` part.

Maybe even more ideal would be to make it possible in one configuration file (on project level) to setup `fixers` by file or directory. This is now not possible, but this would a big BC break.

**Comment by keradus** [comment]
Can't see a conclusion of this issue.
IMO it should be closed.

**Comment by GrahamCampbell** [comment]
:+1: for closing.
